pkg/flowaggregator/certificate.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flowaggregator
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"math/big"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/apiserver/certificate"
+)
+
+const (
+	CAConfigMapName       = "flow-aggregator-ca"
+	CAConfigMapKey        = "ca.crt"
+	CAConfigMapNamespace  = "flow-aggregator"
+	ClientSecretNamespace = "flow-aggregator"
+	// #nosec G101: false positive triggered by variable name which includes "Secret"
+	ClientSecretName = "flow-aggregator-client-tls"
+)
+
+var (
+	validFrom = time.Now().Add(-time.Hour) // valid an hour earlier to avoid flakes due to clock skew
+	maxAge    = time.Hour * 24 * 365       // one year self-signed certs
+)
+
+func generateCACertKey() (*x509.Certificate, *rsa.PrivateKey, []byte, error) {
+	cert := &x509.Certificate{
+		SerialNumber: big.NewInt(2020),
```

Comment:
I think the Serial Number is supposed to be unique for the certificate issuer. So use 1 for the CA, 2 for the Server and 3 for the Client.

Reply:
Updated. Thanks!